src/main/java/com/redhat/vertx/pipeline/PotentiallyRecoverableException.java

```diff
@@ -0,0 +1,4 @@
+package com.redhat.vertx.pipeline;
+
+public interface PotentiallyRecoverableException {
```

Comment:
Shouldn't this be a subclass of RuntimeException?

Reply:
The trick is that we wanted StepDependencyNotMet to be caught (so people know to throw it and be on the lookout for it) and MissingParameterException can't be caught because it happens inside a Map.  The purpose of making a common interface is to simplify checks for either of them.

Reply:
Okay, I understand what you're doing with it now.